docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/src/Controller/CgovNavTreeController.php

```diff
@@ -17,294 +16,469 @@
 class CgovNavTreeController extends ControllerBase {
 
   /**
-   * The path alias manager.
+   * The renderer service.
    *
-   * @var \Drupal\path_alias\AliasManagerInterface
+   * @var \Drupal\Core\Render\RendererInterface
    */
-  protected $aliasManager;
-
-  /**
-   * The Entity Type Manager.
-   *
-   * @var \Drupal\Core\Entity\EntityTypeManager
-   */
-  protected $entityTypeManager;
+  protected $renderer;
 
   /**
    * Constructs a new PathAliasListBuilder object.
    *
-   * @param \Drupal\path_alias\AliasManagerInterface $alias_manager
-   *   The path alias manager.
-   * @param \Drupal\Core\Entity\EntityTypeManager $entityTypeManager
-   *   The Entity Type Manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer.
    */
   public function __construct(
-    AliasManagerInterface $alias_manager,
-    EntityTypeManager $entityTypeManager
+    RendererInterface $renderer
   ) {
-    $this->aliasManager = $alias_manager;
-    $this->entityTypeManager = $entityTypeManager;
+    $this->renderer = $renderer;
   }
 
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('path_alias.manager'),
-      $container->get('entity_type.manager'),
+      $container->get('renderer'),
     );
   }
 
   /**
    * Return section nav items.
    */
   public function getSectionNav(TermInterface $taxonomy_term = NULL) {
```

Comment:
Instead of two functions "getSectionNav" and "getMobileNav" , We can create a single function "getNav" and pass the $menu_type as parameter "section_nav" and "mobile_nav".

Reply:
If you look at the 2 functions you will see that the logic is more than just section_nav or mobile_nav. So to avoid a conditional in a `getNav` function, they have been split up.

Reply:
Ty! I see now why you have split it up. Thanks all looks good